test/compiler/irpasses.jl

```diff
@@ -98,3 +99,13 @@ let nt = (a=1, b=2)
     # Should throw
     @test_throws ArgumentError blah31139(nt)
 end
+
+# Expr(:new) annoted as PartialStruct
+struct FooPartial
+    x
+    y
+end
+@eval f(x) = $(Expr(:new, FooPartial, :x, 2)).x
```

Comment:
why such a weird way to call the constructor?

Reply:
As I said in the commit message, inlining drops the type annotation on the `:new`, so it doesn't break.

Reply:
Sure, but why not just `FooPartial(x) = new(x, 2).x`

Reply:
Ah, sure that'd work I guess. Will do